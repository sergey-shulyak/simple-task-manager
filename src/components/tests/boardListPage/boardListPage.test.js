import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BoardListPage from '../../boardListPage';

const boards = [
  { id: '1', title: 'Test 1' },
  { id: '2', title: 'Test 2' },
  { id: '3', title: 'Test 3' }
];

const props = {
  fetchBoards: jest.fn(),
  showModal: jest.fn(),
  boards
};

describe('BoardListPage', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<BoardListPage {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should call fetchBoards on componentWillMount', () => {
    const wrapper = shallow(<BoardListPage {...props} />);

    wrapper.simulate('componentWillMount');

    expect(props.fetchBoards.mock.calls.length).toBeGreaterThan(0);
  });
});
