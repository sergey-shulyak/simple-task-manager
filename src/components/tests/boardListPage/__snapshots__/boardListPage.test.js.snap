// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardListPage Should render correctly 1`] = `
<div
  className="board-list-page"
>
  <Link
    className="board-list-page__back-link"
    replace={false}
    to="/"
  >
    Go back
  </Link>
  <a
    className="board-list-page__new-board-link"
    onClick={[Function]}
  >
    New board
  </a>
  <BoardList
    className="board-list-page__board-list"
  >
    <BoardEntry
      className="board-list__board"
      description=""
      id="1"
      key="1"
      showModal={[Function]}
      title="Test 1"
    />
    <BoardEntry
      className="board-list__board"
      description=""
      id="2"
      key="2"
      showModal={[Function]}
      title="Test 2"
    />
    <BoardEntry
      className="board-list__board"
      description=""
      id="3"
      key="3"
      showModal={[Function]}
      title="Test 3"
    />
  </BoardList>
</div>
`;
